### ISSUE TITLE
Verify that the client is authenticated clientside

### ISSUE THREAD

**Issue by amxmln** [feathersjs/feathers#1177]
I’m building a feathers app with a vue frontend. I’m trying to protect some of my client-side routes via a navigation guard.

My approach so far was to set a "isLoggedIn" flag to `true` in my Vuex-Store after the initial `app.authenticate()` call (and subesequent login if there was no token in storage) to authentication-client (as suggested by many of the resources I’ve found online), however that doesn’t offer real protection as I’ve found out that the Vuex-Store can be tampered with through the developer-console in the browser. Simply doing a `document.getElementById('app').__vue__.$store.state.user.isLoggedIn = true` is enough to circumvent that protection.

Intuitively I would say calling `app.authenticate()` again if the client is trying to navigate to a protected route should work, however the feathers docs clearly state that `app.authenticate()` with no arguments should only be called once when the application first loads.

So how do I verify that a user is logged in on the client side? Is creating a "checkAuth" service on the server and pinging it the only way, or is there a simpler option I have overlooked?

Thank you for this framework and your time.

**Comment by issy123** [comment]
### direct answer
You can call the app.authenticated call multiple times and use it like a promise to check if the user is logged in or not. 

### problem related answer
Frontend can always be tampered with, they can always have access to secret parts if it is available in the frontend to reach. You should keep your focus on making your app work, and focus on having your backend secured as much as possible.

This means instead of a delete, do a soft-delete to keep track of things when things go wrong. Do a authentication check on the backend for the services and service methods that need those, hide sensitive data like passwords and other attributes you don't want to show.

Also have a look at feathers common hooks. These are hooks that are commonly used with feathers. Like disallowing hooks, hiding attributes, disallowing editing certain attributes etc. 

Goodluck with your application!

**Comment by amxmln** [comment]
Thank you for your reply.

If I can just call the `app.authenticate()` (`app.authenticated` is undefined, so I assume it’s a typo?) method multiple times, why does [this](https://github.com/feathersjs/docs/blob/master/api/authentication/client.md#appauthenticate) warning say it should only be called __once__? If that’s a mistake / no longer relevant, perhaps it should be changed.

I am aware that the client can never be trusted, which was why I was trying to think of ways in which my protection could be circumvented. Obviously, I require authentication for service calls etc. but just being able to _see_ certain areas of the app without authorization so easily was a problem already.

**Comment by amxmln** [comment]
Thank you, I understand how to do it, I’m just hesitant to make multiple calls to `app.authenticate()` when the documentation says it should be called only once.
It would be trivial to create a really small service on the server-side that only allows a `GET` and in the before hook for that simply checks if `params.user` is set and throws a NotAuthenticated-Error if it isn’t. I’m just wondering what the best strategy is in this case since I feel like a lot of people should come across it, but it isn’t really talked about (or maybe I’m just too inexperienced and asking questions that seem self-explanatory to others :sweat_smile: ).

**Comment by issy123** [comment]
I'm glad it's clear. 
I am using it on each new route, as far as i know you can reuse it every time. 

If you do want a clear answer then let's close this issue and re-ask on a simplified way:

*Can you use [app.authenticate()](https://docs.feathersjs.com/api/authentication/client.html#appauthenticate) multiple times to check if the user can navigate to a page that requires login?*

**Comment by amxmln** [comment]
We can do that, yes. Thank you very much for your time and explanations.

**Comment by katwekibs** [comment]
i use app.authenticate() once on login and then use app.reAuthenticate() in my auth route guard.

**Comment by priyabagus** [comment]
If you want a simple check on the client side whether a current user is authenticated or not, you may do this : 

```
import app from './feathers'  

// ...

const isAuthenticated = await app.authentication.authenticated

if(isAuthenticated){
   // do something 
   // ...
}
```
`feathers.js` file
```
// example using socket.io as transport

import feathers from '@feathersjs/feathers'
const socketio = require('@feathersjs/socketio-client')
const io = require('socket.io-client')
const auth = require('@feathersjs/authentication-client')

const app = feathers()
const socket = io('your-server-address')
app.configure(socketio(socket))
app.configure(auth())

export default app
```
